### DISCUSSION TITLE
tables exposed to public?

### DISCUSSION THREAD

**Discussion by man-tax** [Budibase/budibase#5663] [Help] [answered]
Hey,

why is budibase exposing all tables under hostname:10000/app/ta_xyz/rows? That's just my internal network at the moment, but once I reverse proxy to my domain this will be open to the public?! No? What am I missing?

![image](https://user-images.githubusercontent.com/104101494/166009091-68798d44-58b8-483d-8411-a13f59e0a30a.png)

Thanks
Sebastian

**Comment by man-tax** [comment]
All right, so I understand this is the internal API and access depends on the role of the logged in user, as set in 'manage access' in the builder. 

BUT: Say I set access to tables to 'admin', because I don't won't that all logged in users can see all the data. In this case then, 'basic' user can't see any data anymore, right?! So it's either or - either everyone can see everything through the internal api (even though I've set filters on my data providers), OR they can't see anything, because table is restricted to 'admin'?!

Is that really the case?

Thanks
Sebastian

**Comment by mike12345567** [comment, marked as answer]
Hi @man-tax - theres a couple of things to check here:
1. Make sure you're not already logged into Budibase, for example in Chrome if you open the dev-tools and look in "application", then cookies, for your `10.0.0.3:10000` cookie you should see that you have a `budibase:auth` and a `budibase:currentapp` cookie. These cookies combined mean that you are currently authenticated with the platform and therefore can access your tables, the call your making is a basic GET call that builders or basic users can perform if they have access to the app. See the screenshot below for how to see if you have cookie or not. If you try this in an incognito tab you'll find that you don't have access to the data.
![image](https://user-images.githubusercontent.com/4407001/166062321-7eb6814d-b50a-4abd-8055-2109e9c3b3bc.png)
2. Its possible that you have made your table publicly accessible, I believe it is likely the first case, but its possible that you have switched your table to be publicly accessible via the role level selection for the table - in this case you would see the same behaviour.

EDIT: as per your latest comment - right now we don't have granular access control at the API level, this is something that will be worked on in the future. Right now you have to provide access to the whole table to a particular role, you cannot restrict it on a row by row basis, any restrictions made in the data provider are frontend filters and not actual backend restrictions. Granular access control will be coming down the line as we understand it is important however I don't have a solid date/roadmap item to reference at this time.

**Reply by man-tax** [reply]
Thanks a lot! Yes, it's been the first option. Can you have a look at my other question from a minute ago as well? So at the end of the day either all my basic user have access to all data, or to no data at all?

**Reply by mike12345567** [reply]
Hi @man-tax - see my edit to my original comment for response to your other question.

**Reply by man-tax** [reply]
> EDIT: as per your latest comment - right now we don't have granular access control at the API level, this is something that will be worked on in the future. Right now you have to provide access to the whole table to a particular role, you cannot restrict it on a row by row basis, any restrictions made in the data provider are frontend filters and not actual backend restrictions. Granular access control will be coming down the line as we understand it is important however I don't have a solid date/roadmap item to reference at this time.

That's way more than just 'important', even if internal, and a hard blocker for me. Hard to say, having to stop with budibase at this point then. I was able to find a way to secure the attachments, but this isn't going to work :(

**Reply by mike12345567** [reply]
Hi @man-tax - sorry to hear that, I'm sorry I can't give a better estimate as to when this will be implemented - we plan to do some work to our RBAC system in the near future - we just have a very full roadmap at the minute with a lot of core functionality still being put together! 

I'll make sure that the rest of the team is aware of this, as input like this helps us prioritise roadmap items.

**Reply by mjashanks** [reply]
@man-tax - we've had conversations about this one internally a fair bit. Would you be available for a video call sometime soon, so we can understand your needs around this? It would help us scope the feature.

If you're up for helping us out - https://calendly.com/mjashanks/budibase (or email me - check my profile)

Thanks!

**Reply by benaunt** [reply]
Row level security will be an important consideration for my place of work when we venture down this path shortly to. I would assume it would be important for many people looking at the enterprise solution.
